sticker/src/tensorflow/tagger.rs

```diff
@@ -146,12 +146,12 @@ impl TaggerGraph {
 
         let train_op = Self::add_op(&graph, op_names::TRAIN_OP)?;
 
-        let graph_write_op = Self::add_op(&graph, op_names::WRITE_GRAPH_OP)?;
-        let logdir_op = Self::add_op(&graph, op_names::LOGDIR_OP)?;
-        let summary_init_op = Self::add_op(&graph, op_names::SUMMARY_INIT_OP)?;
+        let graph_write_op = Self::add_op(&graph, op_names::WRITE_GRAPH_OP).ok();
```

Comment:
I am not entirely happy about quietly handling all errors in these `add_op` calls as _just-do-not-write-summaries_.  E.g. if a name mismatch is introduced somewhere in python land, we're treating it the same way as a regular 0.4 graph. This would then eventually crash at the `init_logdir` call in `sticker-(pre)train`.

A more verbose option:

~~~rust
let graph_write_op = match Self::add_op(&graph, op_names::WRITE_GRAPH_OP) {
    Ok(op) => op,
    Err(_) => {
        eprintln!("Couldn't find {} in the graph. Summary writing is not supported!\
         This is likely due to a graph from sticker <0.5", op_names::WRITE_GRAPH_OP);
        None
    }
};
~~~

I am also not a fan of that option, since it throws a warning message meant for users who actually want to train at everyone, including probably uninterested tagging-users.

Reply:
Indeed, we should avoid such messages for 0.5 users. There are two other obvious options:

1. Store `Result` rather than an `Option`, so that we still carry the actual errors in case someone tries to use the ops.
2. Use lazy initialization, so that the op is only looked up (once) on use. The overhead should be small, because after initialization only an atomic is used to check whether the value is initialized (a mutex is only used during initialization). This has a questionable benefit over (1), except that the lookup is really only done when necessary.

Reply:
Third option: make a new constructor `new_with_logdir` that is used when the option is passed to `sticker-train`. This would have a small benefit that we could fail in the constructor. Probably makes the constructor even more expensive though.

Reply:
My bad, I didn't realize that we actually have verbose messages in `init_logdir`:

~~~rust
args.add_target(
        self.graph
            .summary_init_op
            .as_ref()
            .ok_or_else(|| err_msg("Graph does not support writing of summaries"))?,
);
~~~

This is the only place where we can expect issues and we are verbose about it, so I think your PR as is should be fine.

Reply:
Ack! Thanks for the review!